lib/common/xml.c

```diff
@@ -1168,62 +1126,72 @@ pcmk__xml_needs_escape(const char *text, bool for_attr)
  *       \c xmlEncodeEntitiesReentrant() and \c xmlEncodeSpecialChars().
  *       However, their escaping is incomplete. See:
  *       https://discourse.gnome.org/t/intended-use-of-xmlencodeentitiesreentrant-vs-xmlencodespecialchars/19252
+ * \note The caller is responsible for freeing the return value using
+ *       \c g_free().
  */
-char *
+gchar *
```

Comment:
For some reason `attribute_{default,text}()` flipped back to using `char*`

Reply:
Whoops, thanks! I shuffled commits around to make the "text newline" fix the last commit. Previously, it was before `pcmk__xml_escape()` changed to `gchar *`.